File: Harpia-Health-Analysis-WS/src/main/java/com/harpia/HarpiaHealthAnalysisWS/model/firebase/FcmToken.java
package com.harpia.HarpiaHealthAnalysisWS.model.firebase;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "firebase_token", uniqueConstraints = @UniqueConstraint(columnNames = "user_id"))
public class FcmToken {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column
    private long id;

    @Column(name = "user_id")
    private int userId;
    @Column
    private String token;

    @Override
    public String toString() {
        return "FcmToken{" +
                "id=" + id +
                ", userId=" + userId +
                ", token='" + token + '\'' +
                '}';
    }
}

File: Flutter-Harpia-Health-Analysis/lib/model/firebase/FcmToken.dart
class FcmToken {
  late int _id;
  late int _patientId;
  late String _token;

  FcmToken({required patientId, required token}) {
    _patientId = patientId;
    _token = token;
  }

  int get id => _id;

  set id(int value) {
    _id = value;
  }

  int get patientId => _patientId;

  String get token => _token;

  set token(String value) {
    _token = value;
  }

  set patientId(int value) {
    _patientId = value;
  }

  @override
  String toString() {
    return 'Token{_id: $_id, _patientId: $_patientId, _token: $_token}';
  }
}
